### ISSUE TITLE
RTL Support Please

### ISSUE THREAD

**Issue by safrilth** [st3v3nmw/obsidian-spaced-repetition#335]
**Is your feature request related to a problem? Please describe.**
I use rtl plugin for some language written from right to left.. it works for editor and preview mode, but not flashcard mode..

**Editor Mode** : 
![editor](https://user-images.githubusercontent.com/54169984/145671494-023e2a0d-f041-411c-9657-372a8fbdab20.PNG)

**Preview / Reading Mode** : 
![Reading](https://user-images.githubusercontent.com/54169984/145671503-2a5e1bb2-1dd9-4f5d-8f1c-a7094367c413.PNG)

**Flashcard Mode** : 
![flashcard](https://user-images.githubusercontent.com/54169984/145671522-36ac6736-8a1e-4ca6-9604-bfc9f926bd24.PNG)

**Describe the solution you'd like**
Flashcard should be display something like Preview / Reading Mode.
the solution can be something to detect if display are using RTL or not (based on frontmatters showed in editor mode above ) : 
if yes, then display flashcard with RTL mode activated

**Comment by ronzulu** [comment]
Shalom and/or salam @danzilberdan 

Just to get some background, I translated a recent comment I posted using google translate:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/098bc190-3b23-43ea-889f-dd4bc0717f17)

However, when I pasted it into Obsidian the RTL didn't match:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/8b74453e-48bc-4c74-964c-1301112552ac)

Then when I changed the Obsidian editor setting for RTL it did display correctly:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/fb3b659e-5672-4f0c-9f9f-a13eec75a255)

Can we assume that this fix only needs to work Obsidian itself is set for RTL?

Are you using the `Obsidian RTL Plugin`. I note that it requires the same frontmatter as per the OP:
```
---
direction: rtl
---
```

Cheers
Ronny

**Comment by danzilberdan** [comment]
Hi!from what I understand, you are

**Comment by danzilberdan** [comment]
Understood. When changing Obsidian's default direction to RTL it works. But it is not useful because I type most of my notes LTR and just some in RTL. I am using Obsidian RTL Plugin and using the frontmatter as you have written.

**Comment by ronzulu** [comment]
I've made a simple change (using the HTML `dir="rtl"` mechanism) which hopefully works.

I've changed yesterday's text to be a question (with Obsidian RTL on)::
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/00e989b3-d54b-4e5b-bda5-c8cb9456b53a)


And Obsidian RTL off:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/875db26d-d189-4598-a265-2ca8beacda78)

The flashcard (I think) displays correctly with this test code (Obsidian RTL off):
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/580013fa-fce7-4c19-95ea-5d3de6ba0512)

Note that I haven't installed the `Obsidian RTL Plugin`

# Testing

Could you please try out this test version of plug-in.
[main_335A.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14079731/main_335A.zip)

You will need to copy that to the obsidian plug-in folder, e.g.
`D:\Obsidian\Obsidian\.obsidian\plugins\obsidian-spaced-repetition`

Then reload the plug-in, or restart obsidian. 

Note that the display of flashcards is currently hard coded for RTL (i.e. doesn't check the direction frontmatter)

**Comment by danzilberdan** [comment]
Perfect! And event better if it would parse the frontmatter. Amazing!

**Comment by ronzulu** [comment]
Great that it worked. I've now added the frontmatter check so hopefully this is complete now.
[main_335B.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14133534/main_335B.zip)

I've also updated the flashcard "quick edit" modal to also honor the frontmatter direction setting, so if you could please verify that that works as well.

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/81e9548b-8f4d-4ad1-beb3-317942324073)

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/8ed108af-c35c-409e-9e27-7e78c38f4ef5)

If all ok, then I'll create the PR so that it can be merged into the next mainstream release.

Cheers
Ronny

**Comment by ronzulu** [comment]
Hi @issam-seghir 

I've noticed that you contributed the Arabic translation to this plug-in and thought you might be interested in the requested RTL functionality that I've implemented.

It's in beta testing at the moment, and was wondering if you would be able to test as well. The latest code is the `B` version in my previous post above.

This new feature is probably only relevant if you have both LTR and RTL flashcards within the one vault, and use the `direction` frontmatter attribute, as per the `Obsidian RTL` Plugin.

Any feedback appreciated.

Cheers
Ronny
